src/akkudoktoreos/config.py

```diff
@@ -1,27 +1,177 @@
+import json
+import os
+import shutil
 from datetime import datetime, timedelta
+from pathlib import Path
+from typing import Any, Optional
 
-output_dir = "output"
-
-prediction_hours = 48
-optimization_hours = 24
-strafe = 10
-moegliche_ladestroeme_in_prozent = [
-    0.0,
-    6.0 / 16.0,
-    7.0 / 16.0,
-    8.0 / 16.0,
-    9.0 / 16.0,
-    10.0 / 16.0,
-    11.0 / 16.0,
-    12.0 / 16.0,
-    13.0 / 16.0,
-    14.0 / 16.0,
-    15.0 / 16.0,
-    1.0,
-]
-
-
-def get_start_enddate(prediction_hours=48, startdate=None):
+from pydantic import BaseModel, ValidationError
+
+EOS_DIR = "EOS_DIR"
+ENCODING = "UTF-8"
+CONFIG_FILE_NAME = "EOS.config.json"
+DEFAULT_CONFIG_FILE = Path(__file__).parent.joinpath("default.config.json")
+
+
+class FolderConfig(BaseModel):
+    "Folder configuration"
+
+    output: str
+    cache: str
+
+
+class EOSConfig(BaseModel):
+    "EOS dependent config"
+
+    prediction_hours: int
+    optimization_hours: int
+    penalty: int
+    available_charging_rates_in_percentage: list[float]
+
+
+class BaseConfig(BaseModel):
+    "The base configuration."
+
+    directories: FolderConfig
+    eos: EOSConfig
+
+
+class AppConfig(BaseConfig):
+    "The app config."
+
+    working_dir: Path
+
+    def run_setup(self) -> None:
+        "Run app setup."
+        print("Checking directory settings and creating missing directories...")
+        for key, value in self.directories.model_dump().items():
+            if not isinstance(value, str):
+                continue
+            path = self.working_dir / value
+            if path.is_dir():
```

Comment:
Should probably omit this check as makedirs will be fine with directory (and maybe fail if it would be a file).

Reply:
I put this check to differentiate the print message. But I can reduce the code and just print the folder